gama/configuration/classification_task/preprocessors.py

```diff
@@ -0,0 +1,288 @@
+    >>>    )
+
+    voila! 🎉 The `gamma` hyperparameter is now added to the Nystroem preprocessor
+    with the shared value available in the `shared_hyperparameters` property.
+
+        How to name my hyperparameters ✍️
+    ------------------------------
+
+    The hyperparameters you add to the config space should be named in the following
+    format if similar hyperparameter names can be found in other preprocessors:
+
+    >>>    <hyperparameter_name>__<preprocessor_name>
+
+    For example, the `gamma` hyperparameter for the `Nystroem` preprocessor should
+    be named `gamma__nystroem` given that the `gamma` hyperparameter is also
+    available in the `RBFSampler` preprocessor. This naming convention is used to ensure
+    that the hyperparameters are added to the correct preprocessor in the config space.
+
+    If your hyperparameter name is unique to your preprocessor, you can name it as you
+    please without the need to have `__<preprocessor_name>` at the end of the name.
+    Nonetheless, following the naming convention would in any way not cause any issues.
+
+    Modify an existing preprocessor 💅
+    -------------------
+
+    To modify an existing preprocessor, adjust its respective setup method and the
+    shared hyperparameters property as needed by modifying the values of the
+    hyperparameters. For example, to change the value of the `gamma` hyperparameter for
+    the `Nystroem` preprocessor, change the value of the `gamma` hyperparameter
+    in the `shared_hyperparameters` property by:
+
+    >>>    "gamma": {"lower": 0.001, "upper": 0.8, "default_value": 0.5},
+
+    The `gamma` hyperparameter will then be added to the config space with the
+    appropriate value. However, be cautious, if you change values in the shared
+    hyperparameters property, it will be changed for all preprocessors that use that
+    hyperparameter. If you want this change to only apply to a specific preprocessor,
+    you should add the hyperparameter to the setup method of that preprocessor.
+    E.g. if you want to change the value of the `gamma` hyperparameter for the
+    `Nystroem` preprocessor, and only want this change to apply to the `Nystroem`
+    preprocessor, add the following line to the `setup_nystroem` method:
+
+    >>>    gamma = csh.CategoricalHyperparameter(
+    >>>        "gamma__nystroem", {"lower": 0.001, "upper": 0.8, "default_value": 0.5},
+    >>>    )
+
+    The `gamma` hyperparameter will be added as-is for the `Nystroem` preprocessor
+    and the value of the `gamma` hyperparameter for other preprocessors will be as
+    available in the `shared_hyperparameters` property – iff they use the `gamma`
+    hyperparameter of the `shared_hyperparameters` property.
+
+
+    Parameters
+    ----------
+    config_space : cs.ConfigurationSpace
+        The ConfigSpace object that will be used to add the preprocessors and their
+        respective hyperparameters.
+
+    """
```

Comment:
Also to docs. Feel free to remove some of the duplication. E.g., if you explain how to define the search space for an algorithm (preprocessor or classifier), you can refer to the earlier explanation if there are no additional caveats. You can always elaborate on specific features like shared hyperparameters.

Reply:
Great ! Will be done in January, following the full review.